### PR TITLE
VSCE: Add CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by abeatrix**
This PR adds the `CONTRIBUTING.md` file for the VS Code extension. 

Closes https://github.com/sourcegraph/sourcegraph/issues/26536

## Test plan

Test plan N/A for Markdown files

### CODE REVIEW

File: client/vscode/CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
+# Contributing to Sourcegraph VS Code Extension
+
+Thank you for your interest in contributing to Sourcegraph!
+The goal of this document is to provide a high-level overview of how you can contribute to the Sourcegraph VS Code Extension.
+_Please refer to our [main CONTRIBUTING](https://github.com/sourcegraph/sourcegraph/blob/main/CONTRIBUTING.md) docs for general information regarding contributing to any Sourcegraph repository._
```
Comment (jjinnii) [inline client/vscode/CONTRIBUTING.md:5]: Super nit: I don't think this sentence needs to be italicized 😄

**Review by jjinnii — APPROVED** [review]
This is amazing, thank you so much!

File: client/vscode/CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+## Development
+
+Please refer to the [Developing Sourcegraph guide](https://docs.sourcegraph.com/dev) to get started.
+
+### Build and run
+
+1. `git clone` the [Sourcegraph repository](https://github.com/sourcegraph/sourcegraph)
+1. Install dependencies via `yarn` for the Sourcegraph repository
+1. Run `yarn generate` at the root directory to generate the required schemas
+1. `cd` to the `client/vscode` directory and run `yarn` and `npm install` to install dependencies for the extension
```
Comment (tjkandala) [inline client/vscode/CONTRIBUTING.md:24]: Can/should we move the 2 vscode specific dependencies to the root level to remove the last step?

File: client/vscode/CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+1. Make your changes to the files within the `client/vscode` directory with VS Code
+1. Run `yarn build` to build or `yarn watch:node` / `yarn watch:web` to build and watch the tasks
+1. Select `Launch VS Code Extension` (`Launch VS Code Web Extension` for VS Code Web) from the dropdown menu to see your changes
```
Comment (tjkandala) [inline client/vscode/CONTRIBUTING.md:27]: Should we call out that this dropdown menu is in the "Run and Debug" sidebar view?

File: client/vscode/CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+### Debugging
+
+Please refer to the [How to Contribute](https://github.com/microsoft/vscode/wiki/How-to-Contribute#debugging) guide by VS Code for debugging tips.
+
+## Questions
+
+If you need guidances or have any questions regarding Sourcegraph or the extension in general, we invite you to connect with us on the [Sourcegraph Community Slack group](https://about.sourcegraph.com/community).
```
Comment (tjkandala) [inline client/vscode/CONTRIBUTING.md:35]:
```suggestion
If you need guidance or have any questions regarding Sourcegraph or the extension in general, we invite you to connect with us on the [Sourcegraph Community Slack group](https://about.sourcegraph.com/community).
```

**Review by tjkandala — APPROVED** [review]
Very nice!

File: client/vscode/CONTRIBUTING.md (hunk 1 above)
Comment (abeatrix) [reply client/vscode/CONTRIBUTING.md:5]: Removed :)

**Comment by abeatrix** [comment]
@tjkandala i'm not sure why 149 files were changed when I only wanted to make changes to the README and CONTRIBUTING files LOL should I just start a new branch and make changes directly to tjk instead or is there a easier way to solve this?

**Comment by tjkandala** [comment]
@abeatrix I'm sure it's because I rebased the base branch again :(. This thread has [some good answers](https://stackoverflow.com/questions/63736743/how-to-rebase-a-branch-after-its-base-branch-was-rebased-on-master). The solution I thought of was to be to rebase onto `tjk/vsce-search-ui`, but then you'd have to resolve the same merge conflicts that I already did (and in the same way!). 

The git docs have a [relevant section](https://git-scm.com/docs/git-rebase#_recovering_from_upstream_rebase) as well.

> Rebasing (or any other form of rewriting) a branch that others have based work on is a bad idea: anyone downstream of it is forced to manually fix their history. This section explains how to do the fix from the downstream’s point of view. The real fix, however, would be to avoid rebasing the upstream in the first place.

I've learned my lesson... fortunately we won't use this feature branch workflow anymore.

> Hard case: The changes are not the same.

Since we had merge conflicts, we're dealing with the [hard case](https://git-scm.com/docs/git-rebase#_the_hard_case). Can you try this out?

**Comment by abeatrix** [comment]
@tjkandala don't worry! I don't think it's worth going through this since it's just two files 😆  I will add this change to my new PR for the feedback sidebar issue that is based on the most updated tjk branch and make a note there if that sounds good to you?

**Comment by abeatrix** [comment]
@tjkandala awesome! It now lives in the bee/vsce-help-sidebar branch where I have also moved the dependencies to the root level, and will continue working on it tomorrow: https://github.com/sourcegraph/sourcegraph/blob/bee/vsce-help-sidebar/client/vscode/CONTRIBUTING.md

**Comment by abeatrix** [comment]
Changes made to the https://github.com/sourcegraph/sourcegraph/pull/32740 instead because of the problem mentioned [above](https://github.com/sourcegraph/sourcegraph/pull/32634#issuecomment-1069325485)

**Pull request closed without merging**